app/models/navigation_link.rb

```diff
@@ -1,10 +1,33 @@
 class NavigationLink < ApplicationRecord
   SVG_REGEXP = /<svg .*>/i.freeze
 
+  before_validation :allow_relative_url, if: :url?
```

Comment:
"Conditional callbacks in code by Michael?" I hear you say?  I'm as surprised as the next person, but the gem we use for URL validation is rather comprehensive and has fewer surprises than `URI.regexp`:

```ruby
URI.regexp(%w[http https]).match?("https://")
#=> true

navigation_link.url = "https://"
#=> "https://"
navigation_link.valid?
#=> false
 navigation_link.errors_as_sentence
#=> "Url is not a valid URL"
```

That's why I decided to just add some pre and post-processing around it instead of writing my own custom validator class (well, technically I did start with the custom validator class, but abandoned that approach quickly).